src/lib/dapp/index.ts

```diff
@@ -208,8 +208,24 @@ export class DAppConnector {
     try {
       const { uri, approval } = await this.connectURI(pairingTopic)
       this.walletConnectModal.openModal({ uri })
-      const session = await approval()
-      await this.onSessionConnected(session)
+
+      const session = await new Promise<SessionTypes.Struct>(async (resolve, reject) => {
```

Comment:
Can we make this an opt-in-param instead?

Reply:
I added the option to `showErrorOnReject` : Whether to show an error when the user rejects the pairing (default: `false`)